core/src/apps/common/request_pin.py

```
@@ -198,6 +291,20 @@
+async def error_pin_used(ctx: wire.Context) -> NoReturn:
+    from trezor.ui.layouts import show_error_and_raise
+
+    await show_error_and_raise(
+        ctx,
+        "warning_wrong_pin",
+        header=_(i18n_keys.PASSPHRASE__PIN_USED),
+        content=_(i18n_keys.PASSPHRASE__PIN_USED_DESC),
+        red=True,
+        exc=wire.PinInvalid,
+    )
+    assert False
```

Ruff (0.12.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)